pkg/validations/cluster.go

```diff
@@ -12,6 +12,42 @@ import (
 	"github.com/aws/eks-anywhere/pkg/types"
 )
 
+// ValidateOSForRegistryMirror checks if the OS is valid for the provided registry mirror configuration.
+func ValidateOSForRegistryMirror(clusterSpec *cluster.Spec) error {
+	cluster := clusterSpec.Cluster
+	if cluster.Spec.RegistryMirrorConfiguration == nil {
+		return nil
+	}
+
+	if !cluster.Spec.RegistryMirrorConfiguration.InsecureSkipVerify ||
+		cluster.Spec.DatacenterRef.Kind == v1alpha1.DockerDatacenterKind {
+		return nil
+	}
+
+	var osFamily v1alpha1.OSFamily
+
+	switch cluster.Spec.DatacenterRef.Kind {
+	case v1alpha1.NutanixDatacenterKind:
+		osFamily = clusterSpec.NutanixMachineConfigs[cluster.Spec.ControlPlaneConfiguration.MachineGroupRef.Name].OSFamily()
```

Comment:
do we support each machine config to have different OS family?

Reply:
yeah, went through some testing to verify and it seems as though we do. I've updated the code to take that into account.

It's more clean now :)